src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,58 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React, { Dispatch, SetStateAction } from 'react';
+import { Filters } from '../../types/Filters';
+type Props = {
+  query: string;
+  setQuery: Dispatch<SetStateAction<string>>;
+  status: Filters;
+  setStatus: Dispatch<SetStateAction<Filters>>;
+};
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+export const TodoFilter: React.FC<Props> = props => {
+  const { query, setQuery, status, setStatus } = props;
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+  return (
+    <form className="field has-addons">
+      <p className="control">
+        <span className="select">
+          <select
+            data-cy="statusSelect"
+            value={status}
+            onChange={event => {
+              setStatus(event.target.value as Filters);
```

Comment:
The `setStatus` function is being passed `event.target.value` directly, which is of type `string`. Since `setStatus` expects a value of type `Filters`, you should ensure that the value is properly cast or converted to the `Filters` type. Consider using a type assertion or a mapping function to convert the string to the appropriate `Filters` enum value.